.github/workflows/build.yaml

```diff
@@ -201,6 +206,155 @@ jobs:
           name: ${{matrix.platform}}-lr-logs
           path: '*-logs.tar.gz'
 
+  libtester-deb-install-test:
+    name: libtester deb install test
+    needs: [d, Build, dev-package]
+    if: always() && needs.dev-package.result == 'success'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu20, ubuntu22]
+    runs-on:  ["self-hosted", "enf-x86-lowtier"] # not sure if this warrants a better machine, but start here
+    container: ${{ matrix.platform == 'ubuntu20' && 'ubuntu:focal' || 'ubuntu:jammy' }}
+    steps:
+      - name: Download cdt
+        uses: AntelopeIO/asset-artifact-download-action@v2
+        with:
+          owner: AntelopeIO
+          repo: cdt
+          file: 'cdt_.*amd64.deb'
+          target: main
+          artifact-name: cdt_ubuntu_package_amd64
+          token: ${{github.token}}
+      - name: Download leap-dev
+        uses: actions/download-artifact@v3
+        with:
+          name: leap-dev-${{matrix.platform}}-amd64
+      - name: Install cdt and leap-dev Packages
+        run: |
+          apt-get update
+          export DEBIAN_FRONTEND='noninteractive'
+          export TZ='Etc/UTC'
+          apt-get install -y ./*.deb
+          rm ./*.deb
+      - name: checkout reference-contracts
+        uses: actions/checkout@v3
+        with:
+          repository: AntelopeIO/reference-contracts
+          path: reference-contracts
+      - name: Build & Test reference-contracts
+        run: |
+          apt-get -y install cmake build-essential
+          cmake -S reference-contracts -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=On
+          cmake --build build -- -j $(nproc)
+          cd build/tests
+          ctest --output-on-failure -j $(nproc)
+
+  libtester-make-dev-install-test:
+    name: libtester make dev-install test
+    needs: [d, Build]
+    if: always() && needs.Build.result == 'success'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu20, ubuntu22]
+    runs-on: ["self-hosted", "enf-x86-hightier"] # not sure if this warrants a different machine, but start here due to dev-install step
+    container: ${{fromJSON(needs.d.outputs.p)[matrix.platform].image}}
+    steps:
+      - name: Clone leap
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Download leap builddir
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{matrix.platform}}-build
+      - name: Extract leap build
+        run: |
+          zstdcat build.tar.zst | tar x
+      - name: leap dev-install
+        run: |
+          cd build
+          ninja install
+          ninja dev-install
```

Comment:
From the logs, it looks as if the entire source tree is getting rebuilt here, which is why you need a high performing runner to get it done in time.

Needs to be refactored somehow to not perform the entire build all over again here.

Reply:
Good catch.  I refactored to `cmake` targets and that looks to have done what I was expecting instead of rebuilding.  Just noticed the `dev-install` target does a build first, which I wasn't expecting.  Also explains why this test was taking so long -- as you noted.

```
cmake --install build
cmake --install build --component dev
```
addressed: https://github.com/AntelopeIO/leap/pull/1259/commits/f069ef42df2860ea749e796d97ccc6c693c18ee5